File: src/main/java/com/pedigree/model/Tag.java
package com.pedigree.model;

import java.util.UUID;

public class Tag {
    private final String id = UUID.randomUUID().toString();
    private String name;

    public String getId() { return id; }
    public String getName() { return name; }
    public void setName(String name) {
        this.name = name;
        com.pedigree.util.DirtyFlag.setModified();
    }
}

File: src/main/kotlin/com/family/tree/model/Tag.kt
package com.family.tree.model

import com.family.tree.util.DirtyFlag
import java.util.UUID

class Tag {
    val id: String = UUID.randomUUID().toString()
    var name: String? = null
        set(value) {
            field = value
            DirtyFlag.setModified()
        }
}
